fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/HeaderNestingJsonBuilderStrategy.java

```diff
@@ -48,6 +49,16 @@ public JsonNode apply(Response response) {
                     DruidJsonResponseContentKeys.STATUS_CODE.getName(),
                     mappingJsonFactory.createParser(String.valueOf(response.getStatusCode())).readValueAsTree()
             );
+            if (CacheFeatureFlag.ETAG.isOn()) {
+                objectNode.set(
+                        DruidJsonResponseContentKeys.ETAG.getName(),
+                        mappingJsonFactory
+                                .createParser(
+                                        response.getHeader(DruidJsonResponseContentKeys.ETAG.getName())
```

Comment:
`If-None-Match` is never present on the response, in addition to it not being present. It the etag related header only present when we receive a 200, there will be no etag related header in response in 300.